src/Confluent.Kafka/Config.cs

```diff
@@ -131,6 +139,10 @@ protected object GetEnum(Type type, string key)
         {
             var result = Get(key);
             if (result == null) { return null; }
+            if (EnumNameToConfigValueSubstitutes.Values.Count(v => v == result) > 0)
```

Comment:
it's getting a little yucky that this is here, but imo it doesn't matter and is appropriate.